lxd/storage/drivers/driver_pure_volumes.go

```diff
@@ -786,7 +948,8 @@ func (d *pure) CreateVolumeSnapshot(snapVol Volume, op *operations.Operation) er
 	revert.Add(func() { _ = d.DeleteVolumeSnapshot(snapVol, op) })
 
 	// For VMs, create a snapshot of the filesystem volume too.
-	if snapVol.IsVMBlock() {
+	// Skip if snapshotVMfilesystem is false to prevent overwriting separately copied volumes.
+	if snapVol.IsVMBlock() && snapshotVMfilesystem {
```

Comment:
is snapshotVMfilesystem a pure specific thing? Why is it needed, seems unusual?

Reply:
This has to do sequential snapshot copying through the main volume, where in such case we do not want to create filesystem volume. Let's talk this one through on the meeting as it will be easier for me to explain.